ingestion-beam/src/main/java/com/mozilla/telemetry/ContextualServicesReporter.java

```diff
@@ -70,31 +72,65 @@ public static PipelineResult run(ContextualServicesReporterOptions.Parsed option
+    // reporting URL
+    // if the count passes a threshold.
+    PCollection<SponsoredInteraction> impressionsCountedByContextId = requests
+        .apply("FilterPerContextIdDocTypes", Filter.by((interaction) -> individualImpressions //
+            .contains(interaction.getDerivedDocumentType())))
+        .apply(LabelSpikes.perContextId(options.getImpressionSpikeThreshold(),
+            Time.parseDuration(options.getImpressionSpikeWindowDuration()),
+            TelemetryEventType.IMPRESSION));
+
     // Aggregate impressions.
-    PCollection<SponsoredInteraction> aggregated = requests
-        .apply("FilterAggregatedDocTypes", Filter.by((interaction) -> aggregatedDocTypes //
+    PCollection<SponsoredInteraction> aggregatedGenuineImpressions = requests
+        .apply("FilterAggregatedDocTypes", Filter.by((interaction) -> individualImpressions //
             .contains(interaction.getDerivedDocumentType())))
+        .apply("FilterForLegitImpressions",
+            Filter.by(new SerializableFunction<SponsoredInteraction, Boolean>() {
+
+              @Override
+              public Boolean apply(SponsoredInteraction input) {
+                return !input.getReportingUrl().contains("impression-status=1");
+              }
+            }))
         .apply(AggregateImpressions.of(options.getAggregationWindowDuration()));
 
-    // Perform windowed click counting per context_id, adding a click-status to the reporting URL
-    // if the count passes a threshold.
-    PCollection<SponsoredInteraction> perContextId = requests
-        .apply("FilterPerContextIdDocTypes", Filter.by((interaction) -> perContextIdDocTypes //
+    PCollection<SponsoredInteraction> aggregatedPossiblyFraudulentImpressions = requests
+        .apply("FilterAggregatedDocTypes", Filter.by((interaction) -> individualImpressions //
             .contains(interaction.getDerivedDocumentType())))
-        .apply(LabelClickSpikes.perContextId(options.getClickSpikeThreshold(),
-            Time.parseDuration(options.getClickSpikeWindowDuration())));
+        .apply("FilterForPossiblyFraudulentImpressions",
+            Filter.by(new SerializableFunction<SponsoredInteraction, Boolean>() {
+
+              @Override
+              public Boolean apply(SponsoredInteraction input) {
+                return input.getReportingUrl().contains("impression-status=1");
+              }
+            }))
+        .apply(AggregateImpressions.of(options.getAggregationWindowDuration()));
 
     PCollection<SponsoredInteraction> unaggregated = requests.apply("FilterUnaggregatedDocTypes",
         Filter.by((interaction) -> !unionedDocTypes //
             .contains(interaction.getDerivedDocumentType())));
 
-    PCollectionList.of(aggregated).and(perContextId).and(unaggregated).apply(Flatten.pCollections())
+    PCollectionList.of(aggregatedGenuineImpressions).and(aggregatedPossiblyFraudulentImpressions)
```

Comment:
If we combine the two aggregated impression sets into one i think this should just be:

```java
PCollecitonList.of(aggregatedImpressions).and(clicksCountedByContextId).and(unaggregated)
  .apply(Flatten.pCollections())
```